selftests/functional/test_basic.py

```diff
@@ -554,9 +554,8 @@ def test_output_skip(self):
         self.assertIn('skiponsetup.py:SkipOnSetupTest.test_wont_be_executed:'
                       '  SKIP', result.stdout)
 
+    @unittest.skipIf(missing_binary('echo'), 'echo binary not available')
     def test_ugly_echo_cmd(self):
-        if not os.path.exists("/bin/echo"):
-            self.skipTest("Program /bin/echo does not exist")
         os.chdir(basedir)
         cmd_line = ('./scripts/avocado run "/bin/echo -ne '
```

Comment:
(the later requires the use of `self.skipTest` unless you want to run the same command twice)